src/intrinsics/node/buffer.js

```diff
@@ -0,0 +1,135 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+/* @flow */
+
+import invariant from "../../invariant.js";
+import { Realm } from "../../realm.js";
+import { Value, NumberValue, NativeFunctionValue, ObjectValue, StringValue } from "../../values/index.js";
+import { Set, ToInteger } from "../../methods/index.js";
+
+declare var process: any;
+
+function getBufferFromArg(realm: Realm, value: Value): Uint8Array {
+  invariant(value instanceof ObjectValue && value.$ViewedArrayBuffer);
+  let buffer = value.$ViewedArrayBuffer;
+  invariant(buffer instanceof ObjectValue && buffer.$ArrayBufferData);
+  return buffer.$ArrayBufferData;
+}
+
+export default function (realm: Realm): ObjectValue {
+  let nativeBuffer = process.binding('buffer');
+  let nativeBufferPrototype = (require('buffer'): any).Buffer.prototype;
```

Comment:
Hm. This is defined in native C code. This is related to the `setupBufferJS` thing below. The runtime we're currently running in has already called `setupBufferJS`, read from a mutated object and that's how it gets a handle on this object.

https://github.com/nodejs/node/blob/v7.9.0/src/node_buffer.cc#L1197-L1230

We can't do the same thing at this point. Luckily this same object is reachable another way through another module which I found here.